src/object-store/src/cache_policy.rs

```diff
@@ -31,19 +30,41 @@ use crate::metrics::{
     OBJECT_STORE_LRU_CACHE_MISS,
 };
 
+#[derive(Clone)]
 pub struct LruCacheLayer<C> {
     cache: Arc<C>,
     lru_cache: Arc<Mutex<LruCache<String, ()>>>,
 }
 
-impl<C: Accessor> LruCacheLayer<C> {
-    pub fn new(cache: Arc<C>, capacity: usize) -> Self {
-        Self {
+impl<C: Accessor + Clone> LruCacheLayer<C> {
+    pub async fn new(cache: Arc<C>, capacity: usize) -> Result<Self> {
+        let zelf = Self {
```

Comment:
I think we could just use `layer` as the variable name. I only see the usage of `zelf` in our python script crate.